### PR TITLE
React Capstone Project

### PR DESCRIPTION
**Pull request by SaboorSohaib**
**Hi Dear!** 

### I implemented the following requirements.

- Use React documentation.
- Use React components.
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write integration tests with a React testing library.
- Use styles in a React app.
- Use React life cycle methods.
- Apply React best practices and language style guides in code.
- Use store, actions and reducers in React.

### - Video Link:(https://www.loom.com/share/3f99ac9dc2064e34bd01717ffd7c0bfc)

### CODE REVIEW

File: src/components/home.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      <div className="title-div">
+        <h3 className="title">See All Countries</h3>
+      </div>
+      <div className="list-page">
+        <AllCountry data={country} />
+      </div>
+    </div>
+  );
+};
+
```
Comment (DammyShittu) [inline src/components/home.js]: - [ ] Your homepage is required to have search or filtering functionality. There should be a way for searching items on the home page. For example, if I type in a country name or any data type of your choosing the home page should display only items that match the search data. Kindly implement that.

  furkan-filter-2

File: README.md [hunk 2]
```diff
@@ -244,7 +205,7 @@ Feel free to check the [issues page](../../issues/).
 
 > Write a message to encourage readers to support your project
 
-If you like this project...
+If you like this project you can give star and fork it.
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (DammyShittu) [inline README.md:209]: - [ ] The "Acknowledgment" section of your readme documentation should contain credit given to Nelson Sakwa whose design was used. Kindly add that.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -35,33 +35,34 @@ After you're finished please remove all the comments and instructions!
 
 # 📗 Table of Contents
```
Comment (FrederickMih) [inline README.md:34]: - [**Optional**] There's a Microverse logo above the Table of Contents, kindly replace it with your project's logo or remove it. Also, remove the text(**Microverse README Template**).

File: README.md [hunk 4]
```diff
@@ -35,33 +35,34 @@ After you're finished please remove all the comments and instructions!
...
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [❓ FAQ ](#-faq-)
+  - [📝 License ](#-license-)
 
 <!-- PROJECT DESCRIPTION -->
 
-# 📖 [your_project_name] <a name="about-project"></a>
+# 📖 Europe Countries Map <a name="about-project"></a>
 
-> Describe your project in 1 or 2 sentences.
```
Comment (FrederickMih) [inline README.md:62]: - [Optional] Kindly remove the entire blockquote in your README file. There are directives that give you advice on how to customize your README file. In the end, you need to remove them.

File: src/components/home.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from 'react';
+import { useSelector, shallowEqual, useDispatch } from 'react-redux';
+import { fetchCountry } from '../redux/countries/fetchcountries';
+import AllCountry from './allCountries';
+import europe from '../images/europe.png';
+import '../App.css';
+
+let retriveData = true;
+const HomePage = () => {
```
Comment (FrederickMih) [inline src/components/home.js:9]: - [Optional] We are expected to follow the design guidelines for this project including typography, and layout. Kindly compare the two screenshots below and make the requested changes.
- It is important to note that having a nice design increases your chances for a job as recruiters will be checking your app.

| Yours | Guidelines |
| --- | --- |
| Screenshot from 2022-12-07 14-03-40 | Screenshot from 2022-12-02 19-27-09 |

Comment (FrederickMih) [inline src/components/home.js:9]: - [ ] As requested by the previous reviewer and as stipulated in the requirements of this project, the Home page should have a list of categories that could be filtered using a given parameter, e.g. country name. We are expected to filter data retrieved from the API using a **Filter** stateless component. I think you'll require a search field on the Home page like the image below to achieve this. Filtering should probably be done using the ES6 filter() method.
Screenshot from 2022-12-07 14-16-21
NB. The image is just for demonstrative purposes.

- As you can see, a user needs to scroll through the entire page to select a country right below which is very tedious and does not give a good user experience while using your app. Let's be able to filter data using the search field.

File: src/components/singleCountry.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable */
```
Comment (FrederickMih) [inline src/components/singleCountry.js:1]: - [**Optional**] It's not a good practice of disabling Linters. These are great tools that monitor your codebase and flag programmatic/stylistic errors. It is very important to fix these issues.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @SaboorSohaib , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 

- Commit messages are capitalized :+1: 
- All Linter checks are passing :heavy_check_mark: 


## Suggested changes

_Check the comments under the review._

## Optional Suggested :+1: 

[**Optional**]Your app should have integration tests using React Testing Library. You also need to test all your components and pure functions in the Redux store.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Davidosky007 — APPROVED** [review]
Hi @SaboorSohaib 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Nice looking readme file 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**